src/crypto-providers/trezorCryptoProvider.ts

```diff
@@ -729,7 +746,7 @@ const TrezorCryptoProvider: () => Promise<CryptoProvider> = async () => {
 
     validateVotingRegistrationAddressType(addressParams.addressType)
```

Comment:
The `getAddressParams` functions above is a bit shoddy and not fully compatible with this usage. An address with staking controlled by script (or even one with third-party staking key!) won't be processed correctly here.